napari/_qt/menus/file_menu.py

```diff
@@ -225,8 +68,23 @@ def _add_sample(*args, plg=plugin_name, smp=samp_name):
                             stack=False,
                         )
 
-                menu.addAction(action)
-                action.triggered.connect(_add_sample)
+                display_name = samp_dict['display_name'].replace("&", "&&")
+                if multiprovider:
+                    title = display_name
+                else:
+                    title = menu_item_template.format(
+                        plugin_name, display_name
+                    )
+                action = Action(
+                    id=samp_dict['id'],
+                    title=title,
+                    menus=[{'id': sub_menu_id, 'group': MenuGroup.NAVIGATION}],
+                    callback=_add_sample,
+                )
+
+                Q_FILE_ACTIONS.append(action)
```

Comment:
still not sure we want to be keeping this in this file? We never construct a `FileMenu` afaict so I don't even think we're running this? We should be able to totally get rid of this file.

Reply:
File removed!